services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -162,14 +163,25 @@ public void run() {
       try {
         TopicState topicState = topicState();
         if (topicState == null) {
+          if(_topicCreationEnabled) {
```

Comment:
You probably needs to think about how to make the logic in `run()` clean and easy to read.

For example, `topicState.currentReplicationFactor()` is called 3 times. I am not if they are all needed. If topic auto create is not enabled and topic is not created, it doesn't make sense to simply do `return` in `run()`.